jf_agent/agent_logging.py

```diff
@@ -60,22 +71,105 @@ def emit(self, record) -> None:
         _emit_helper(self, record, always_use_newlines=True)
 
 
-def configure(outdir, debug_requests=False):
-
+class CustomQueueListener(QueueListener):
+    _jf_sentinel = -1
+
+    def handle(self, record: Union[LogRecord, int]) -> None:
+        if record is self._jf_sentinel:
+            for handler in self.handlers:
+                handler.handle(record)
+                return
+        return super().handle(record)
+
+
+class CustomQueueHandler(QueueHandler):
+    def __init__(self, queue: Queue[Any], webhook_base: str, api_token: str) -> None:
+        super().__init__(queue)
+        self.webhook_base = webhook_base
+        self.api_token = api_token
+        self.messages_to_send = []
+        self.last_message_send_time = datetime.now()
+        # Indication that logging has finished and we should send whatever is left in the current batch.
+        self._sentinel = -1
+        self.initiated_at = datetime.strftime(datetime.now(), '%Y-%m-%d-%H-%M-%S')
+        self.create_stream = True
+
+    def post_logs_to_jellyfish(self, now: datetime) -> bool:
+        headers = {'Content-Type': 'application/json', 'X-JF-API-Token': self.api_token}
+        url = self.webhook_base + '/agent-logs'
+        with temp_disable_request_loggers():
+            resp = requests.post(
+                url,
+                json={'logs': self.messages_to_send, 'create_stream': self.create_stream},
+                headers=headers,
+            )
+        if resp.ok:
+            self.messages_to_send = []
+            self.last_message_send_time = now
+            if self.create_stream and resp.json()['stream_created']:
+                self.create_stream = False
+
+    def handle(self, record: Union[logging.LogRecord, int]) -> None:
+        now = datetime.now()
+
+        if record is not self._sentinel:
```

Comment:
Nit: In this instance, better practice to use `==`. Generally should reserve `is` for actually checking that two objects have the same identity (e.g. are the same object), rather than checking equality.

See here for a good explanation on this: https://stackoverflow.com/a/25758019